### DISCUSSION TITLE
exporting sample genotypes including the imputed alleles

### DISCUSSION THREAD

**Discussion by chlangley** [tskit-dev/tskit#2115] [Q&A]
I want to export the tsinferred sample genotypes with the imputations for missing data.
ts.write_fasta() is not the ideal format.
This is no doubt a familiar path.  Please direct me.

Cheers,
Chuck

**Comment by jeromekelleher** [comment]
Is VCF any more useful? ``tskit vcf <ts file>`` should do it from the command line if you want a quick look. Otherwise the python [write_vcf](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence.write_vcf) will give you more control of the output.

**Comment by chlangley** [comment]
> On 3/Feb/2022, at 7:04 AM, Jerome Kelleher ***@***.***> wrote:
> 
> 
> Is VCF any more useful? tskit vcf <ts file> should do it from the command line if you want a quick look. Other wise the python write_vcf will give you more control of the output.


OK.  But, Ben wrote recently:


> Begin forwarded message:
> 
> From: Ben Jeffery ***@***.***>
> Subject: Re: [tskit-dev/tskit] Imputation in Treesequence.write_vcf() ? (Discussion #2078)
> Date: January 5, 2022 at 4:31:48 AM PST
> To: tskit-dev/tskit ***@***.***>
> Cc: chlangley ***@***.***>, Author ***@***.***>
> Reply-To: tskit-dev/tskit ***@***.***>
> 
> 
> Hi @chlangley!
> The write_vcf code doesn't yet support missing data, and will error if asked to output it. The work to support missing data isn't currently scheduled, but we're always open to contributions if you'd like to have a go!
> 
> (I'm moving this to the tskit repo as that's where this code lives)
>

**Reply by jeromekelleher** [reply]
I agree that is confusing!

However, I *think* that the output of tsinfer won't contain any missing data because it has been imputed. If I remember properly, the input sample data is marked as missing data, but we don't try to maintain "missingness" in the inferred output and there's no missing data in the output VCF. 

@hyanwong @awohns @szhan, is this correct?

**Reply by hyanwong** [reply]
All the sites are imputed, so the resulting vcf shouldn't contain any missing data. We intend to have a post-processing step to "add back in" some sorts of missingness: see https://github.com/tskit-dev/tsinfer/issues/224

**Reply by chlangley** [reply]
So I return to my question.  
What's the most accessible process for capturing the samples genotypes, including the imputed alleles at positions in which particular samples had missing data (-1)?  

Does ts.write_fasta() output the imputed alleles?

**Reply by chlangley** [reply]
Sorry. I misread your post. 
the vcf output of the samples should have the imputed alleles.
OK.  That's the path then.  Thanks.
